frontend/src/components/TicketList.js

```diff
@@ -22,11 +23,32 @@ const TicketList = ({ allTickets }) => {
               ]
     )
 
-    if (
-        Object.keys(TICKET_TYPES)
-            .map((type) => allTickets[type])
-            .flat().length === 0
-    ) {
+    useEffect(() => {
+        if (
+            Object.keys(TICKET_TYPES)
+                .map((type) => allTickets[type])
+                .flat().length === 0
+        ) {
+            setIsLoading(true)
```

Comment:
I dont like how we assume that we're in a loading state just because there are zero tickets, I would pass the loading state from Dashboard.js, and add a ticketsLoading state in Dashboard as well